Segment-Optimizely-X/Classes/SEGOptimizelyXIntegration.m

```diff
@@ -0,0 +1,178 @@
+        NSString *segmentAnonymousId = [self.analytics getAnonymousId];
+        BOOL trackKnownUsers = [[self.settings objectForKey:@"trackKnownUsers"] boolValue];
+
+        if (trackKnownUsers && [self.userId length] == 0) {
+            SEGLog(@"Segment will only track users associated with a userId when the trackKnownUsers setting is enabled.");
+            return;
+        }
+
+        if (trackKnownUsers) {
+            if (self.userTraits.count > 0) {
+                [self.client track:payload.event userId:self.userId attributes:self.userTraits eventTags:payload.properties];
+                SEGLog(@"[optimizely track:%@ userId:%@ attributes:%@ eventTags:%@]", payload.event, self.userId, self.userTraits, payload.properties);
+            } else {
+                [self.client track:payload.event userId:self.userId eventTags:payload.properties];
+                SEGLog(@"[optimizely track:%@ userId:%@ eventTags:%@]", payload.event, self.userId, payload.properties);
+            }
+        }
+
+
+        if (self.userTraits.count > 0) {
+            [self.client track:payload.event userId:segmentAnonymousId attributes:self.userTraits eventTags:payload.properties];
+            SEGLog(@"[optimizely track:%@ userId:%@ attributes:%@ eventTags:%@]", payload.event, segmentAnonymousId, self.userTraits, payload.properties);
+        } else {
+            [self.client track:payload.event userId:segmentAnonymousId eventTags:payload.properties];
+            SEGLog(@"[optimizely track:%@ userId:%@ eventTags:%@]", payload.event, segmentAnonymousId, payload.properties);
+        }
+    }
+}
+
+- (void)reset
+{
+    if ([self.manager getOptimizely] == nil) {
+        return;
+    } else {
+        [[NSNotificationCenter defaultCenter] removeObserver:self.observer];
+        SEGLog(@"[NSNotificationCenter defaultCenter] removeObserver:%@", self.observer);
+    }
+}
+
+#pragma mark - Experiement Viewed For NSNotification
+
+- (void)experimentDidGetViewed:(NSNotification *)notification
+{
+    OPTLYExperiment *experiment = notification.userInfo[OptimizelyNotificationsUserDictionaryExperimentKey];
+    OPTLYVariation *variation = notification.userInfo[OptimizelyNotificationsUserDictionaryVariationKey];
+    NSDictionary *properties = @{
+        @"experimentId" : [experiment experimentId],
+        @"experimentName" : [experiment experimentKey],
+        @"variationId" : [variation variationId],
+        @"variationName" : [variation variationKey]
+    };
+
+    // Trigger event as per our spec https://segment.com/docs/spec/ab-testing/
+    [self.analytics track:@"Experiment Viewed" properties:properties];
+    SEGLog(@"[[SEGAnalytics sharedAnalytics] track:@'Experiment Viewed' properties:%@", properties);
+}
+
+#pragma mark - Private - Queueing
+
+- (void)enqueueAction:(SEGTrackPayload *)payload
+{
+    SEGLog(@"%@ Optimizely not initialized. Enqueueing action: %@", self, payload);
+    @try {
+        if (self.queue.count > 1000) {
```

Comment:
Should we set a condition to stop queuing events? Should set timer? Android has 5 minutes before stopping